### PR TITLE
Added test for sslMode=prefer #43

### PR DESCRIPTION
**Pull request by Mart-Bogdan**
Add new test case

### CODE REVIEW

**Comment by Mart-Bogdan** [comment]
Hello, @Narigo this is additional test for PR #88

File: src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java [hunk 1]
```diff
@@ -106,4 +106,53 @@ public void testNoSslConfiguration(TestContext context) {
...
+        .put("port", Integer.parseInt(System.getProperty("dbssl.port", "54321")))
+        .put("sslMode", "prefer")
+    );
+
+    System.out.println("testPreferSslConfiguration");
+
+    clientSsl.getConnection(sqlConnectionAsyncResult -> {
+      context.assertTrue(sqlConnectionAsyncResult.succeeded());
+      conn = sqlConnectionAsyncResult.result();
+      System.out.println("testPreferSslConfiguration step2");
```
Comment (Narigo) [inline src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java]: We could get rid of these debug messages I think

File: src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java [hunk 2]
```diff
@@ -106,4 +106,53 @@ public void testNoSslConfiguration(TestContext context) {
...
+      conn.query("SELECT 1", ar -> {
+        System.out.println("testPreferSslConfiguration callback2");
```
Comment (Narigo) [inline src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java]: We could get rid of these debug messages I think

File: src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java [hunk 3]
```diff
@@ -106,4 +106,53 @@ public void testNoSslConfiguration(TestContext context) {
...
+        if (ar.failed()) {
+          context.fail("Should not fail on ssl connection");
+        } else {
+          System.out.println("testPreferSslConfiguration SSL OK");
+
+          clientNoSsl.getConnection(sqlConnectionAsyncResult2 -> {
+            context.assertTrue(sqlConnectionAsyncResult2.succeeded());
+            conn = sqlConnectionAsyncResult2.result();
+            System.out.println("testPreferSslConfiguration step3");
```
Comment (Narigo) [inline src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java]: We could get rid of these debug messages I think

File: src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java [hunk 4]
```diff
@@ -106,4 +106,53 @@ public void testNoSslConfiguration(TestContext context) {
...
+            conn.query("SELECT 1", ar2 -> {
+              System.out.println("testPreferSslConfiguration callback4");
```
Comment (Narigo) [inline src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java]: We could get rid of these debug messages I think

File: src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java [hunk 5]
```diff
@@ -106,4 +106,53 @@ public void testNoSslConfiguration(TestContext context) {
...
+        .put("sslMode", "prefer")
+    );
+    SQLClient clientNoSsl = createClient(vertx,
+      new JsonObject()
+        .put("host", System.getProperty("db.host", "localhost"))
+        .put("port", Integer.parseInt(System.getProperty("dbssl.port", "54321")))
+        .put("sslMode", "prefer")
+    );
+
+    System.out.println("testPreferSslConfiguration");
```
Comment (Narigo) [inline src/test/java/io/vertx/ext/asyncsql/PostgreSQLSslConfigurationTest.java]: We could get rid of these debug messages I think

**Pull request merged**